controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -199,6 +216,27 @@ func (r *StorageClassRequestReconciler) reconcilePhases() (reconcile.Result, err
 				break
 			}
 		}
+
+		// check if a cephblockpool resource exists for the desired storageconsumer and storageprofile.
+		if r.cephBlockPool.Name == "" {
+			cephBlockPoolList := &rookCephv1.CephBlockPoolList{}
+			listOptions := &client.MatchingLabels{
+				controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
+				controllers.StorageProfileSpecLabel:  storageProfile.GetSpecHash(),
+			}
+			if err := r.list(cephBlockPoolList, client.InNamespace(r.OperatorNamespace), listOptions); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// if we find only one CephBlockPool with our query, we're good
+			// if we find more than one, we can't determine which one to select, so error out
```

Comment:
just curious,
1. which scenarios will result in more than one blockpool? (apart from the current bug)
2. if there's a 1 (profile):n (pool), what action should be taken? delete one of the pools manually (if at all possible)

asking because, values for both the labels are constant once created, correct me if I'm wrong.